scripts/junit2issue.py

```diff
@@ -0,0 +1,206 @@
+                        'ZEPHYR_VERSION' : version,
+                        'RESULT' : res.message,
+                        'PLATFORM': suite.name,
+                        'LOGS': res.text,
+                    }
+                    report_list.append(report_case)
+                    # print("  " + res.text)
+    return report_list
+
+def gen_content_from_template(template, context):
+    '''
+        gen_content_from_template
+        @template: in template string
+        @context: in context
+        return:
+            generated issue content string
+    '''
+    _s = Template(template)
+    content = _s.substitute(context)
+    # logging.info(content)
+    return content
+
+def github_login():
+    '''
+        github_login
+    '''
+
+    access_token = os.getenv('GITHUB_ACCESS_TOKEN')
+
+    if not access_token:
+        print("no access token find in system environment variable\n")
+        print("please set GITHUB_ACCESS_TOKEN with:\n")
+        print("export GITHUB_ACCESS_TOKEN=<your github access token>\n")
+        return None
+
+    # using an access token
+    return Github(access_token)
+
+def check_issue_existing(repository, report):
+    '''
+        check whether the report issue exists
+        @repository: in repo target
+        @report: dict fo reporting content
+        return:
+            existing issue or None
+    '''
+    tod = datetime.datetime.now()
+    d = datetime.timedelta(days=60)
+    a = tod - d
+    open_issues = repository.get_issues(state='open', since=a, labels=['bug'])
+    for _issue in open_issues:
+        if report['CASE_NAME'] in _issue.title:
+            return _issue
+        if report['LOGS'] and report['LOGS'] in _issue.body:
+            return _issue
+
+    return None
+
+def get_desc_template():
+    '''
+        get_desc_template
+    '''
+    m = imp.load_source('GITHUB_TEMPLETE', 'github_issue_template.tmpl')
+    return m.GITHUB_TEMPLETE
+
+def get_title_template():
+    '''
+        get_title_template
+    '''
+    m = imp.load_source('GITHUB_ISSUE_TITLE_TEMPLETE', 'github_issue_template.tmpl')
+    return m.GITHUB_ISSUE_TITLE_TEMPLETE
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.DEBUG,
+                        format='[%(filename)s:%(lineno)d] %(threadName)s %(message)s')
+
+    auto_report = False
+    parser = argparse.ArgumentParser(description='Process parameters')
+    parser.add_argument('-v', '--vfolder', dest='vfolder', help='version folder')
+    parser.add_argument('-i', '--pfile', dest='pfile', help='platform report file', required=True)
+    parser.add_argument('-a', '--auto_report', dest='auto_report', help='auto report')
+    parser.add_argument('-r', '--repo', dest='report_repo', help='report repo', required=True)
+
+    args = parser.parse_args()
+
+    logging.info(args)
+    pfile = args.pfile
+    vfolder = args.vfolder
+    report_repo = args.report_repo
+
+    xml_file = os.path.join('..\\results', vfolder, pfile)
```

Comment:
`xml_file = os.path.join('..', 'results', vfolder, pfile)`

'..\\\\results' doesn't work under linux (based on os you can have \ or /).

Reply:
updated